events/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProvider.java

```diff
@@ -51,49 +56,123 @@ public void clear(String realmId, long olderThan) {
+    static EventEntity convertEvent(Event event) {
+        EventEntity eventEntity = new EventEntity();
+        eventEntity.setId(UUID.randomUUID().toString());
+        eventEntity.setTime(event.getTime());
+        eventEntity.setType(event.getType().toString());
+        eventEntity.setRealmId(event.getRealmId());
+        eventEntity.setClientId(event.getClientId());
+        eventEntity.setUserId(event.getUserId());
+        eventEntity.setSessionId(event.getSessionId());
+        eventEntity.setIpAddress(event.getIpAddress());
+        eventEntity.setError(event.getError());
         try {
-            e.setDetailsJson(mapper.writeValueAsString(o.getDetails()));
+            eventEntity.setDetailsJson(mapper.writeValueAsString(event.getDetails()));
         } catch (IOException ex) {
             logger.error("Failed to write log details", ex);
         }
-        return e;
-    }
-
-    static Event convert(EventEntity o) {
-        Event e = new Event();
-        e.setTime(o.getTime());
-        e.setType(EventType.valueOf(o.getType()));
-        e.setRealmId(o.getRealmId());
-        e.setClientId(o.getClientId());
-        e.setUserId(o.getUserId());
-        e.setSessionId(o.getSessionId());
-        e.setIpAddress(o.getIpAddress());
-        e.setError(o.getError());
+        return eventEntity;
+    }
+
+    static Event convertEvent(EventEntity eventEntity) {
+        Event event = new Event();
+        event.setTime(eventEntity.getTime());
+        event.setType(EventType.valueOf(eventEntity.getType()));
+        event.setRealmId(eventEntity.getRealmId());
+        event.setClientId(eventEntity.getClientId());
+        event.setUserId(eventEntity.getUserId());
+        event.setSessionId(eventEntity.getSessionId());
+        event.setIpAddress(eventEntity.getIpAddress());
+        event.setError(eventEntity.getError());
         try {
-            Map<String, String> details = mapper.readValue(o.getDetailsJson(), mapType);
-            e.setDetails(details);
+            Map<String, String> details = mapper.readValue(eventEntity.getDetailsJson(), mapType);
+            event.setDetails(details);
         } catch (IOException ex) {
             logger.error("Failed to read log details", ex);
         }
-        return e;
+        return event;
+    }
+    
+    static AdminEventEntity convertAdminEvent(AdminEvent adminEvent) {
+        AdminEventEntity adminEventEntity = new AdminEventEntity();
+        adminEventEntity.setId(UUID.randomUUID().toString());
+        adminEventEntity.setTime(adminEvent.getTime());
+        setAuthDetails(adminEventEntity, adminEvent.getAuthDetails());
+        adminEventEntity.setOperationType(adminEvent.getOperationType().toString());
+        adminEventEntity.setResourcePath(adminEvent.getResourcePath());
+        adminEventEntity.setError(adminEvent.getError());
+        
+        try {
+            adminEventEntity.setRepresentation(mapper.writeValueAsString(adminEvent.getDetails()));
```

Comment:
The representation field of AdminEvent is set incorrectly. It will be re factored and set to the json representation of the created or updated entity/object.